File: tests/truss1.cpp
#include <iostream>
#include <vector>

#include "node.h"
#include "element.h"
#include "case.h"
#include "vtkWriter.h"


int main(int argc, char **argv) {
    parafem::NodePtr n1 (new parafem::Node(-1, 0, 0));
    parafem::NodePtr n2 (new parafem::Node(0, 0, 0));
    parafem::NodePtr n3 (new parafem::Node(1, 0, 0));
    n1->fixed = parafem::Vector3(0,0,0);
    n3->fixed = parafem::Vector3(0,0,0);
    n2->external_force.y() -= 1;
    std::shared_ptr<parafem::TrussMaterial> mat (new parafem::TrussMaterial(1000));
    mat->d_structural = 0.00;
    parafem::TrussPtr t1 (new parafem::Truss(parafem::NodeVec{n1, n2}, mat));
    parafem::TrussPtr t2 (new parafem::Truss(parafem::NodeVec{n2, n3}, mat));
    parafem::FemCasePtr c1 (new parafem::FemCase(parafem::ElementVec{t1, t2}));
    parafem::VtkWriter writer = parafem::VtkWriter("/tmp/parafem/truss1/output");
    for (int i=0; i<1000; i++)
    {
        c1->explicit_step(0.01);
        if (i % 10 == 0)
            writer.writeCase(c1);
        
    }
}

File: tests/truss1.py
import parafem
import numpy as np

mat = parafem.TrussMaterial(1000)
mat.rho = 1
mat.d_structural = 0.0
mat.d_velocity = 1

node1 = parafem.Node(-1, 0, 0)
node2 = parafem.Node(0, 0, 0)
node3 = parafem.Node(1, 0, 0)

node1.fixed = np.array([0, 0, 0])
node3.fixed = np.array([0, 0, 0])
node2.add_external_force(np.array([0, 1, 0]))

truss1 = parafem.Truss([node1, node2], mat)
truss2 = parafem.Truss([node2, node3], mat)

case = parafem.Case([truss1, truss2])

writer = parafem.vtkWriter("/tmp/parafem/truss1_py/output")

for i in range(10000):
    case.explicit_step(0.01)
    if (i % 10) == 0:
        writer.writeCase(case, 0.3)
